examples/peft_llm_gnn/llm_gnn_model.py

```diff
@@ -59,9 +61,34 @@ def get_lm_node_feats(g, lm_feat_names, lm_ntypes):
 
     return lm_feats
 
+class DummyNodeInputLayer(nn.Module):
+    def __init__(self):
+        super(DummyNodeInputLayer, self).__init__()
 
-class LLMGraphModel(gsmodel.GSgnnNodeModelBase):
-    """ LLMGNN Model Class
+    def require_cache_embed(self):
+        return False
+    
+    def forward(self, input_feats, input_nodes):
+        return input_feats
+
+class CosineRandomNegative(nn.Module):
+    def __init__(self):
+        super(CosineRandomNegative, self).__init__()
+        self.margin = 0.5
+
+    def forward(self, pos_score, neg_score):
+        pos_distance = []
+        neg_distance = []
+        for k in pos_score:
```

Comment:
do you iterate over each element here? why not implement it in a vector way?

Reply:
It's a dict of {node_type: scores} as output by the LinkPredictDotDecoder